controllers/servers/host_definer/storage_manager/host_definer_server.py

```diff
@@ -1,13 +1,47 @@
+from controllers.array_action.errors import HostNotFoundError, HostAlreadyExists
+from controllers.array_action.storage_agent import detect_array_type, get_agent
 from controllers.common.csi_logger import get_stdout_logger
+from controllers.common.node_info import NodeIdInfo
 from controllers.servers.host_definer.types import DefineHostResponse
+from controllers.servers.utils import join_object_prefix_with_name, get_array_connection_info_from_secrets
 
 logger = get_stdout_logger()
 
 
 class HostDefinerServicer:
+    def define_host(self, request):  # pylint: disable=invalid-name
+        node_id_info = NodeIdInfo(request.node_id)
+        initiators = node_id_info.initiators
+        connectivity_type = request.connectivity_type
 
-    def define_host(self, request):
-        return DefineHostResponse()
+        host_name = join_object_prefix_with_name(prefix=request.prefix, name=node_id_info.node_name)
+        logger.debug("host name : {}".format(host_name))
 
-    def undefine_host(self, request):
-        return DefineHostResponse()
+        try:
+            array_connection_info = get_array_connection_info_from_secrets(request.system_info)
```

Comment:
```suggestion
            array_connection_info = get_array_connection_info_from_secrets(request.secret)
```

Reply:
its not the k8s secret, its the [system_info](https://github.com/IBM/ibm-block-csi-driver/blob/53ee1f5c850eddf5255f26873b609f04255c24b3/controllers/servers/csi/controller_types.py#L7) ([like](https://github.com/IBM/ibm-block-csi-driver/blob/53ee1f5c850eddf5255f26873b609f04255c24b3/controllers/servers/host_definer/watcher/watcher_helper.py#L196) - don't know why not the same..) object that [get_agent](https://github.com/IBM/ibm-block-csi-driver/blob/53ee1f5c850eddf5255f26873b609f04255c24b3/controllers/array_action/storage_agent.py#L71) get

Reply:
ok, wrong PR